packages/salesforcedx-vscode-soql/src/queryResultsView/media/queryDataViewController.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+(function() {
+  const vscode = acquireVsCodeApi();
+
+  function updateUIWith(queryData, documentName) {
+    const title = document.getElementById('webview-title');
+    title.innerText = documentName;
+
+    renderTableWith(queryData);
+  }
+
+  function loadState() {
+    const state = vscode.getState();
+    if (state) {
+      updateUIWith(state.data, state.documentName);
+    }
+  }
+
+  loadState();
+
+  function renderTableWith(tableData) {
+    new Tabulator('#data-table', {
+      data: tableData,
+      autoColumns: true,
+      pagination: 'local',
+      paginationSize: 10,
+      layout: 'fitColumns',
+      maxHeight: '100%'
+    });
+  }
```

Comment:
There are a quite few ways to customize and control behavior of the rendered table. We can leverage these as need presents itself. For now, I started with a basic set up.

http://tabulator.info/docs/4.8/options